queue.go

```diff
@@ -63,26 +89,21 @@ func (eq *Queue) Close() error {
 }
 
 // run is the main goroutine to flush events to the target sink.
-func (eq *Queue) run() {
+func (eq *Queue) run(ctx context.Context) {
```

Comment:
Need to hook in context into the goroutine's lifecycle here. Will require a major refactor of this implementation though.